sockeye/inference.py

```diff
@@ -353,6 +369,26 @@ def length_ratio_mean(self) -> float:
     def length_ratio_std(self) -> float:
         return self.config.config_data.data_statistics.length_ratio_std
 
+def load_source_factor_vocabs(model_folder: str) -> List[Dict[str,int]]:
+    """
+    Loads source factor vocabs of the form "vocab.src.N" until one can no longer be found.
+    This could alternately be done by first reading the number of factors from the config file.
```

Comment:
I would suggest only trying to load as many vocabs as you expect from the config.